packages/Telephony/Actions/BatchDTMF.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property
+    /// </summary>
+    public class BatchDTMF : Dialog
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Telephony.BatchDTMF";
+        private const string AggregationDialogMemory = "dialog.aggregation";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BatchDTMF"/> class.
+        /// </summary>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public BatchDTMF([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the regex pattern to use to decide when the dialog has aggregated the whole message.
+        /// </summary>
+        [JsonProperty("terminationConditionRegexPattern")]
+        public StringExpression TerminationConditionRegexPattern { get; set; }
```

Comment:
Naming: would `TerminationConditionRegex` do the trick? Not sure if pattern adds a lot and makes the name pretty lengthy. Brainstorming here, would love to hear thoughts.